File: Stack_Queue/queue.cpp
#include <stdio.h>
#include <queue>

using namespace std;

int main(int argc, char const *argv[])
{
    /* Initialize the queue */
    queue<int> queue;

    /* Enqueue elements into the queue */
    queue.push(1);
    queue.push(3);
    queue.push(2);
    queue.push(5);
    queue.push(4);

    /* Access the front element of the queue */
    int front = queue.front();
    printf("Front element is %d\n", front);

    /* Dequeue an element from the queue */
    queue.pop(); // No return value
    printf("Front element is %d\n", queue.front());

    /* Get the length of the queue */
    int size = queue.size();

    /* Check if the queue is empty */
    bool empty = queue.empty();
    return 0;
}

File: java/queue/queue.java
import java.util.LinkedList;
import java.util.Queue;

public class queue {
  public static void main(String[] args) {
    /* Initialize the queue */
    Queue<Integer> queue = new LinkedList<>();

    /* Enqueue elements */
    queue.offer(1);
    queue.offer(3);
    queue.offer(2);
    queue.offer(5);
    queue.offer(4);

    /* Access the first element */
    int peek = queue.peek();

    /* Dequeue an element */
    int pop = queue.poll();

    /* Get the length of the queue */
    int size = queue.size();

    /* Check if the queue is empty */
    boolean isEmpty = queue.isEmpty();
  }
}
